lib/core/osal/netdev_funcs.c

```diff
@@ -252,11 +284,28 @@ netdev_get_channels(const char *ifname)
 
     eth_channels.cmd = ETHTOOL_GCHANNELS;
     ifr.ifr_data     = (void *)&eth_channels;
-    if (ioctl(fd, SIOCETHTOOL, &ifr) < 0)
-        goto err;
+    ret              = ioctl(fd, SIOCETHTOOL, &ifr);
+    if (ret) {
+        if (errno != EOPNOTSUPP) {
+            ret = -errno;
+            goto err;
+        }
+        /* If the device says it has no channels,
+         * try to get rx tx from sysfs, otherwise all traffic
+         * is sent to a single stream, so max queues = 1.
+         */
+        uint32_t rx_count, tx_count;
```

Comment:
Should these variables be set to zero or some know value?

Reply:
yeah I can initialize them